### PR TITLE
Added GetItems of GRPC server and client

### PR DESCRIPTION
**Pull request by stasBigunenko**
#15 
Added GetItems of GRPC server and client, still working on testing

### CODE REVIEW

File: pkg/grpc/serverGRPC.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	defer cancel()
+
+	if in.Username == "" {
+		return &pb.Items{}, status.Error(codes.InvalidArgument, "invalid username")
+	}
+	items := []*pb.Obj{}
+
+	data, err := s.storage.GetItems(ctx, in.Username)
+	if err != nil {
+		return &pb.Items{}, status.Error(codes.Internal, "internal problem 000000000")
```
Comment (romanilchyshyn) [inline pkg/grpc/serverGRPC.go]: `000000000` is this error code?

File: pkg/grpc/serverGRPC_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package grpc
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"monorepa/service/items"
+
+	//"github.com/stretchr/testify/assert"
```
Comment (romanilchyshyn) [inline pkg/grpc/serverGRPC_test.go]: commented imports should be omitted

File: pkg/grpc/serverGRPC_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	go serveBufconn(ln, d)
+	client := makeBufconnClient(ln)
+
+	x := []storage.Storage{
+		{"295bb267-122e-4ab7-a0a4-851490f98095", "XVLBZGBAICMRAJWW", "  fdzdgrxomvt ler"},
+	}
+
+	getItems, _ := client.GetItems(context.Background(), "I")
+
+	assert.Equal(t, x[0].Title, getItems[0].Title, "Everything is good")
```
Comment (romanilchyshyn) [inline pkg/grpc/serverGRPC_test.go]: `"Everything is good"` this should be err message, not success

File: pkg/storage/storage.go [hunk 4]
```diff
@@ -1 +1,63 @@
 package storage
+
+import (
+	"context"
+	"errors"
+	"github.com/google/uuid"
+	"math/rand"
+)
+
+type Storage struct {
```
Comment (romanilchyshyn) [inline pkg/storage/storage.go]: this could be `StorageItem` but not `Storage` for sure

File: pkg/storage/storage.go [hunk 5]
```diff
@@ -1 +1,63 @@
...
+
+func NewStorage() *Storage {
+	return &Storage{}
+}
+
+const letterBytes = "abcdefgh ijklmnop qrstuvw xyz"
+const letterBytesUpper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+// RandStringBytesfunction generate lower runes and spaces
+func RandStringBytes(n int) string {
```
Comment (romanilchyshyn) [inline pkg/storage/storage.go]: we don't want to export names used internally only

File: pkg/storage/storageInterface.go [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package storage
+
+import "context"
+
+type StorageInterface interface {
```
Comment (romanilchyshyn) [inline pkg/storage/storageInterface.go]: it is a good practice not to use `Interface` suffix in the names

File: service/items/clientGRPC.go [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+// Client GRPC
+
+type GRPCClient struct {
+	Client pb.GrpcServiceClient
+}
+
+func New() *GRPCClient {
+	gc := &GRPCClient{}
+
+	conn, _ := grpc.Dial(":8080", grpc.WithInsecure())
```
Comment (romanilchyshyn) [inline service/items/clientGRPC.go]: addr should be configurable

File: pkg/storage/storage.go [hunk 8]
```diff
@@ -1 +1,63 @@
...
+func RandStringBytesUpper(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytesUpper[rand.Intn(len(letterBytesUpper))]
+	}
+	return string(b)
+}
+
+// GetItems generate slice of Storage according the username and send back
+func (d *Storage) GetItems(c context.Context, u string) ([]Storage, error) {
```
Comment (romanilchyshyn) [inline pkg/storage/storage.go]: have no idea what is going on here. method on `Storage` return slice of `Storage`-s

File: pkg/storage/storage.go [hunk 9]
```diff
@@ -1 +1,63 @@
...
+	items := []Storage{}
+
+	if u == "" {
+		return nil, errors.New("invalid user name")
+	}
+
+	for i := range u {
+
+		id := uuid.New()
+		title := RandStringBytesUpper(len(id))
```
Comment (romanilchyshyn) [inline pkg/storage/storage.go]: I think there is no need to complicate implementation and `uuid.New()` for `title` & `description` is pretty enough

File: pkg/grpc/serverGRPC.go (hunk 1 above)
Comment (stasBigunenko) [reply pkg/grpc/serverGRPC.go]: changed to internal problem

File: pkg/grpc/serverGRPC_test.go (hunk 2 above)
Comment (stasBigunenko) [reply pkg/grpc/serverGRPC_test.go]: deleted all unused imports

File: pkg/grpc/serverGRPC_test.go (hunk 3 above)
Comment (stasBigunenko) [reply pkg/grpc/serverGRPC_test.go]: changed to "need to fix test program"

File: pkg/storage/storage.go (hunk 4 above)
Comment (stasBigunenko) [reply pkg/storage/storage.go]: implemented StorageItem and Storage struct

File: pkg/storage/storage.go (hunk 5 above)
Comment (stasBigunenko) [reply pkg/storage/storage.go]: changed to internal

File: pkg/storage/storageInterface.go (hunk 6 above)
Comment (stasBigunenko) [reply pkg/storage/storageInterface.go]: changed name on StorageItemService

File: service/items/clientGRPC.go (hunk 7 above)
Comment (stasBigunenko) [reply service/items/clientGRPC.go]: Added configurable string

File: pkg/storage/storage.go (hunk 9 above)
Comment (stasBigunenko) [reply pkg/storage/storage.go]: left the same code, but can change it if it is needed

**Comment by romanilchyshyn** [comment]
Looks like we don't need `GRPCClient` anymore because of implemented one in https://github.com/stasBigunenko/monorepa/pull/30. 
I'd recommend to make it unexposed and use only for testing purposes.

**Comment by stasBigunenko** [comment]
> Looks like we don't need `GRPCClient` anymore because of implemented one in #30.
> I'd recommend to make it unexposed and use only for testing purposes.

Done with new commit https://github.com/stasBigunenko/monorepa/pull/21/commits/e944abb3cfe65aba3183cff2d9565b54eed93fce

**Review by romanilchyshyn — APPROVED** [review]
LGTM

File: pkg/grpc/clientGRPC.go [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+	"monorepa/pkg/storage"
+)
+
+// Client GRPC
+
+type gRPCClient struct {
+	client pb.GrpcServiceClient
+}
+
+func new(addr string) *gRPCClient {
```
Comment (DrIhor) [inline pkg/grpc/clientGRPC.go]: Where you use it???

Comment (stasBigunenko) [reply pkg/grpc/clientGRPC.go]: It was used only for testing

Comment (DrIhor) [reply pkg/grpc/clientGRPC.go]: Don`t see any test with it. Look for unused code.

Comment (stasBigunenko) [reply pkg/grpc/clientGRPC.go]: deleted func new with the last commit

**Review by DrIhor — APPROVED** [review]

**Pull request merged**